sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -296,6 +296,21 @@ object SQLConf {
       .longConf
       .createWithDefault(250 * 1024 * 1024)
 
+  object HiveCaseSensitiveInferenceMode extends Enumeration {
+    type HiveCaseSensitiveInferenceMode = Value
+    val INFER_AND_SAVE, INFER_ONLY, NEVER_INFER = Value
+  }
+  val HIVE_CASE_SENSITIVE_INFERENCE = buildConf("spark.sql.hive.caseSensitiveInferenceMode")
+    .doc("Configures the action to take when a case-sensitive schema cannot be read from a Hive " +
+      "table's properties. Valid options include INFER_AND_SAVE (infer the case-sensitive " +
```

Comment:
Probably want to add some explanation as to why this matters even though SQL is not case sensitive, e.g.
// Note that Spark SQL itself not case-sensitive, but since file formats such as parquet are, a case-sensitive schema is required in order to properly read fields.